src/pages/ConciergePage.tsx

```diff
@@ -25,19 +29,39 @@ type ConciergePageProps = ConciergePageOnyxProps & StackScreenProps<AuthScreensP
  *     - Else re-route to the login page
  */
 function ConciergePage({session}: ConciergePageProps) {
+    const styles = useThemeStyles();
+    const isUnmounted = useRef(false);
+
     useFocusEffect(() => {
         if (session && 'authToken' in session) {
+            App.confirmReadyToOpenApp();
             // Pop the concierge loading page before opening the concierge report.
             Navigation.isNavigationReady().then(() => {
-                Navigation.goBack(ROUTES.HOME);
-                Report.navigateToConciergeChat();
+                if (isUnmounted.current) {
+                    return;
+                }
```

Comment:
We return early here if the component is unmounted to prevent the user from being navigated to the concierge chat even after closing the concierge page